src/Sylius/Bundle/ApiBundle/ApiPlatform/StateProcessor/MessengerProcessor.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Sylius Sp. z o.o.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\ApiBundle\ApiPlatform\StateProcessor;
+
+use ApiPlatform\Metadata\Operation;
+use ApiPlatform\State\ProcessorInterface;
+use Sylius\Bundle\ApiBundle\StateProcessor\Input\InputDataProcessorInterface;
+
+final readonly class MessengerProcessor implements ProcessorInterface
+{
+    /** @param InputDataProcessorInterface[] $inputDataProcessors */
+    public function __construct(
+        private ProcessorInterface $decoratedProcessor,
+        private iterable $inputDataProcessors,
+    ) {
+    }
+
+    public function process(mixed $data, Operation $operation, array $uriVariables = [], array $context = []): mixed
+    {
+        foreach ($this->inputDataProcessors as $inputDataProcessor) {
+            if ($inputDataProcessor->supports($data, $operation, $uriVariables, $context)) {
+                [$data, $operation, $uriVariables, $context] = $inputDataProcessor->process($data, $operation, $uriVariables, $context);
+            }
+        }
```

Comment:
I have no problems with merging it as is, but this part needs to be rethought, right now it smells.
We're using APIP's processor interface (for no good reason to be honest) that returns `mixed` but are always expecting a 4-element array with specific types.
IMO either providing a custom interface with a DTO or finding another way to get this logic in place would be better. Maybe even reverting completely to how it's done in APIP by default, though I hate those decoration chains.

Reply:
Soo, let's not block this PR. @TheMilek or @NoResponseMate can you create a task to rethink this part?